### PR TITLE
[DPTOOLS-733] airflow pools CLI remove undefined existing pools

### PR DESCRIPTION
**Pull request by milton0825**
When you run
```
airflow pool -f file/path/to/yaml
```
- existing airflow pools not defined in yaml will be removed.
- airflow pools defined in yaml will be added if not exist.
- pool properties changes will be reflected.

### CODE REVIEW

**Comment by milton0825** [comment]
@feng-tao @ArgentFalcon @astahlman

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -259,39 +259,63 @@ def pools(args):
     :param args: parsed args object with a property of filepath to airflow pools yaml file
     :return: None
     """
+    session = settings.Session()
     with open(args.filepath, 'r') as stream:
         try:
             pools_config = yaml.load(stream)
         except yaml.YAMLError:
+            print("Failed to load: {}".format(args.filepath))
```
Comment (astahlman) [inline airflow/bin/cli.py]: Not directly related to your PR, but just noticed that we swallow this exception. Should we? (I don't see why we should)

File: tests/core.py [hunk 2]
```diff
@@ -62,6 +62,7 @@
 from airflow.configuration import AirflowConfigException
 
 import six
+import tempfile
```
Comment (astahlman) [inline tests/core.py]: Doesn't look like this is used?

File: tests/core.py [hunk 3]
```diff
@@ -1053,6 +1054,61 @@ def setUp(self):
...
+
+        cli._pools(session, pools_config)
+
+        self.assertEqual('pool1', session.add.call_args_list[0][0][0].pool)
+
+    def test_pools_delete_pool_if_not_defined(self):
+        session = mock.MagicMock()
+        session.delete = mock.MagicMock()
+        query = session.query.return_value
+        query.all.return_value = [
```
Comment (astahlman) [inline tests/core.py]: I like these tests, but there's a lot of boilerplate related to setting up the mocks that we could abstract out. Maybe something like this?

```
def _expect_session_returns(self, existing_pools):
  session = mock.MagicMock()
  session.delete = mock.MagicMock()
  session.add = mock.MagicMock()
  query = session.query.return_value
  query.all.return_value = existing_pools
  return session

def _assert_pool_was_deleted(self, pool_name)
  self.assertEqual(pool_name, session.delete.call_args_list[0][0][0].pool)
```

And now the test becomes highly readable:
```
def test_pools_delete_pool_if_not_defined(self):
  session = self._expect_session_returns(Pool(pool='pool1'...))
  cli._pools(session, {})
  self._assert_pool_was_deleted('pool1')
```

**Comment by feng-tao** [comment]
👍  lgtm

File: airflow/bin/cli.py (hunk 1 above)
Comment (milton0825) [reply airflow/bin/cli.py]: yeah we should pop the error up.

File: tests/core.py (hunk 2 above)
Comment (milton0825) [reply tests/core.py]: removed.

File: tests/core.py (hunk 3 above)
Comment (milton0825) [reply tests/core.py]: Agree that tests should also be readable

File: airflow/bin/cli.py [hunk 4]
```diff
@@ -259,39 +259,63 @@ def pools(args):
...
-        session.add(
-            Pool(
-                pool=pool_name,
-                slots=int(pool_slots_and_description.get('slot_counts')),
-                description=pool_slots_and_description.get('description'))
-        )
+            print("No need to update pool: {}".format(existing_pool.pool))
+
+    for pool_to_update in pools_to_update:
+        print("Need to update pool: {}".format(pool_to_update))
```
Comment (ArgentFalcon) [inline airflow/bin/cli.py]: nit: `Need to add pool`

File: airflow/bin/cli.py (hunk 1 above)
Comment (ArgentFalcon) [reply airflow/bin/cli.py]: Sometimes it's nice to swallow exception on the CLI and print nice statements instead.

Comment (ArgentFalcon) [reply airflow/bin/cli.py]: Yeah. `exit(n)` where `n` is a value greater than 0 will do that though.

File: airflow/bin/cli.py [hunk 5]
```diff
@@ -259,39 +259,63 @@ def pools(args):
     :param args: parsed args object with a property of filepath to airflow pools yaml file
     :return: None
     """
+    session = settings.Session()
     with open(args.filepath, 'r') as stream:
         try:
             pools_config = yaml.load(stream)
-        except yaml.YAMLError:
+        except yaml.YAMLError as e:
+            print("Failed to load: {}".format(args.filepath))
```
Comment (astahlman) [inline airflow/bin/cli.py]: If all we're doing is printing the exception, I don't think we need this except block at all. The exception should already get logged if it bubbles up

**Pull request merged**